ddtrace/internal/datadog/profiling/stack_v2/src/stack_renderer.cpp

```diff
@@ -119,6 +130,13 @@ StackRenderer::render_python_frame(std::string_view name, std::string_view file,
     if (!utf8_check_is_valid(file.data(), file.size())) {
         file = invalid;
     }
+    // DEV: Echion pushes a dummy frame containing task name, and its line
+    // number is set to 0.
+    if (!task_state.pushed_task_name and line == 0 and name != invalid and name != "<invalid>") {
```

Comment:
How do you feel about skipping the checks for invalid names?  We already propagate invalid symbol and filenames.

Actually, the UX for that might not be great, since it'll erroneously associate a bunch of execution into potentially invalid tasks (like we might have hundreds of tasks all associated to "<invalid>").

Instead:  what if we have a static sequence number that (atomically) increments every time we see an invalid task, and name the task something like "invalid-XXX"?  That'd eliminate the need to check, it'd make sure tasks are labeled when execution is associated to a task (this is important!), and it'd isolate execution away from a common label.

Reply:
There's always the option of "fixing" echion to report better metadata. For instance, we could make the frame object carry a `kind`, that we could set to a value representing "Task", instead of working around the shortcoming?

Reply:
Let me try removing those checks. 

@P403n1x87's suggestion would be the cleanest way here. Propagate a flag from echion and check that instead of relying on `line == 0`. I intentionally avoided changing echion code as I didn't want it to diverge more.

Reply:
Rather than propagating new information in the underlying struct, it might be cleaner to structure the interface to push a different category of information.

We'd have a `render_task_name` interface in the echion fork's Render system.  The Echion implementation of Render would just push it as a frame, and ddtrace would push it into ddup as a task name.

Reply:
Yeah, we should update echion. I'm just delaying such changes to happen after we merge our fork to the upstream echion.